htlcswitch/link.go

```diff
@@ -1150,6 +1228,132 @@ func (l *channelLink) htlcManager() {
 				"PendingLocalUpdateCount")
 		}
 
+		// If we are attempting to shutdown the link to cooperatively
+		// close and the channel is clear of our pending updates and we
+		// haven't already sent a Shutdown, we'll send one now.
+		if (l.shutdownInit || l.shutdownReceived) && !l.shutdownSent &&
+			l.channel.PendingLocalUpdateCount() == 0 {
+
+			// Create the Shutdown we'll send over. Send the local
+			// upfront shutdown script if it exists, and the
+			// provided delivery address otherwise. This is used in
+			// case we did not initiate the coop close.
+			upfrontAddr := l.channel.LocalUpfrontShutdownScript()
```

Comment:
Couldn't we use `chooseDeliveryScript` here rather? Simplifies needing to re-check the upfront shutdown again.

Reply:
not sure what you mean here?

Reply:
I don't think we can do that because if we were to call `chooseDeliveryScript(l.channel.LocalUpfrontShutdownScript(), l.cfg.DeliveryAddr)` it will return an error due to mismatch if the upfront script exists since `l.cfg.DeliveryAddr` will always be set